pkg/providers/aws/cluster_network_provider_test.go

```diff
@@ -1061,6 +1061,56 @@ func TestNetworkProvider_CreateNetwork(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "verify ec2 VpcLimitExceeded returns an error",
+			fields: fields{
+				Client: fake.NewFakeClientWithScheme(scheme, buildTestInfra()),
+				RdsApi: buildMockRdsClient(nil),
+				Ec2Api: buildMockEc2Client(func(ec2Client *mockEc2Client) {
+					ec2Client.vpcs = buildValidClusterVPC(defaultNonOverlappingCidr)
+					ec2Client.describeSubnetsFn = func(input *ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error) {
+						return &ec2.DescribeSubnetsOutput{
+							Subnets: buildStandaloneSubnets(),
+						}, nil
+					}
+					ec2Client.createVpcFn = func(input *ec2.CreateVpcInput) (*ec2.CreateVpcOutput, error) {
+						return &ec2.CreateVpcOutput{}, awserr.New("VpcLimitExceeded", "The maximum number of VPCs has been reached.", nil)
+					}
+				}),
+				ElasticacheApi: buildMockElasticacheClient(nil),
+				Logger:         logrus.NewEntry(logrus.StandardLogger()),
+			},
+			args: args{
+				ctx:  context.TODO(),
+				CIDR: buildValidCIDR(validCIDRSixteen),
+			},
+			wantErr: true,
+		},
+		{
+			name: "verify ec2 InvalidVpcRange returns an error",
+			fields: fields{
+				Client: fake.NewFakeClientWithScheme(scheme, buildTestInfra()),
+				RdsApi: buildMockRdsClient(nil),
+				Ec2Api: buildMockEc2Client(func(ec2Client *mockEc2Client) {
+					ec2Client.vpcs = buildValidClusterVPC(defaultNonOverlappingCidr)
+					ec2Client.describeSubnetsFn = func(input *ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error) {
+						return &ec2.DescribeSubnetsOutput{
+							Subnets: buildStandaloneSubnets(),
+						}, nil
+					}
+					ec2Client.createVpcFn = func(input *ec2.CreateVpcInput) (*ec2.CreateVpcOutput, error) {
+						return &ec2.CreateVpcOutput{}, awserr.New("InvalidVpcRange", "The maximum number of VPCs has been reached.", nil)
```

Comment:
Shouldn't the message match the one we receive from [aws](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html#api-error-codes-table-client)? :thinking: (`The specified CIDR block range is not valid. The block range must be between a /28 netmask and /16 netmask`)

Reply:
Woops

Reply:
Fixed